internal/io/http/rest_sink.go

```diff
@@ -64,6 +79,13 @@ func (ms *RestSink) collectWithUrl(ctx api.StreamContext, item interface{}, desU
 		return fmt.Errorf("rest sink decode data error: %v", err)
 	}
 
+	if ms.compressor != nil {
+		decodedData, err = ms.compressor.Compress(decodedData)
+		if err != nil {
+			return err
+		}
+	}
+
```

Comment:
Http header needs to be set when using compression like `req.Header.Set("Content-Encoding", "gzip")`

Reply:
the Content-Encoding header seems to only set in the response rather than request